pkg/controllers/gateway/gateway_controller.go

```diff
@@ -351,6 +375,96 @@ func selectClusters(gateway gatewayv1beta1.Gateway) []string {
 	return []string{}
 }
 
+func (r *GatewayReconciler) DNSDeletion(ctx context.Context, gateway gatewayv1beta1.Gateway) error {
+	trafficAccessor := traffic.NewGateway(&gateway)
+	allHosts := trafficAccessor.GetHosts()
+
+	for _, host := range allHosts {
+		managedZone, subDomain, err := r.Host.GetManagedZoneForHost(ctx, host, trafficAccessor)
+		if err != nil {
+			return err
+		}
+		if managedZone == nil {
+			log.Log.Info("Managed zone not found cannot get DNS record")
+			return nil
+		}
+
+		log.Log.Info("getting DNS record associated with deleting gateway")
+		dnsRecord, err := r.Host.GetDNSRecord(ctx, subDomain, managedZone)
+		if err != nil && !k8serrors.IsNotFound(err) {
```

Comment:
would there be value in adding a ```r.Host.DeleteDNSForHost()``` that takes does this piece of logic (IE get the managed zone? and deletes the DNSRecord? rather than having the gateway controller have to understand and do that?

Reply:
I feel like line 392 - 405 could become part of the reusable DNS service. At some point we would like to add support an "ingress" gateway so it will also need to do these types of things

Reply:
Ok yes if we are planning on doing this again that makes sense